src/opti-lib/src/OptiEnProfitSubject.cpp

```diff
@@ -144,6 +146,21 @@ double OptiSubjectEnProfit::GetVerbose(const EnjoLib::Matrix & dataMat, bool ver
         //const double pentalityUndervolted = load < 0 ? GMat().Fabs(load * load * load) : 0;
         const double pentalityUndervolted = battery.num_undervolted;
         const double pentalityOvervolted = battery.num_overvolted;
+        
+        if (not sys.buying)
+        {
+            if (pentalityUndervolted > 0)
+            {
+                unacceptableSolution = true;
```

Comment:
@endorxmr : That idea of yours, where we treat buyng/selling separately, allowed to speed up the calculations by another 50%